microcloud/service/microceph.go

```diff
@@ -131,10 +133,33 @@ func (s CephService) Join(ctx context.Context, joinConfig JoinConfig) error {
 }
 
 // ClusterMembers returns a map of cluster member names and addresses.
-func (s CephService) ClusterMembers(ctx context.Context) (map[string]string, error) {
-	client, err := s.Client("", "")
-	if err != nil {
-		return nil, err
+// Optionally sends a remote request using the information in ServerInfo.
+func (s CephService) ClusterMembers(ctx context.Context, info *mdns.ServerInfo) (map[string]string, error) {
```

Comment:
Same here, I think it would be clearer to have a separate function when we want to use the remote client.